### PR TITLE
ENH: pick_events() adapted to step events

### PR DESCRIPTION
**Pull request by cmoutard**

### CODE REVIEW

**Comment by agramfort** [comment]
can you give an example of what you have in mind?

you'll need to add a test and make travis happy afterwards

**Comment by cmoutard** [comment]
# Example

Let's say I have a data set called raw. I am extracting the events from it with the step option which gives the following:

```
>>> events_step = mne.find_events(raw, output = 'step')
Trigger channel contains negative values. Taking absolute value.
484 events found
Events id: [    0     5     6     9    10    17    18    33    34 32768]

>>> events_step[:10]
array([[222225,      0,     10],
       [222229,     10,      0],
       [223476,      0,     18],
       [223480,     18,      0],
       [224726,      0,     34],
       [224730,     34,      0],
       [228862,      0,  32768],
       [229030,  32768,      0],
       [237048,      0,      5],
       [237055,      5,      0]])
```

Now I want to exclude from the events list all the events 32768, but still keeping the step format...
With the present pick_event() function, I can either use the option *include*:

```
>>> events_step = mne.pick_events(events_step, include = [0, 5, 6, 9, 10, 17, 18, 33, 34])
```
Or the option *exclude*:

```
>>> events_step = mne.pick_events(events_step, exclude = [32768])
```

In both cases, the output is:

```
>>> events_step[:10]
array([[222225,      0,     10],
       [222229,     10,      0],
       [223476,      0,     18],
       [223480,     18,      0],
       [224726,      0,     34],
       [224730,     34,      0],
       [229030,  32768,      0],
       [237048,      0,      5],
       [237055,      5,      0],
       [238282,      0,     33]])
```

The row  [229030, 32768, 0]  is unwillingly still present as any other similar rows with the format  [Time,  32768, 0].
My PR aims at solving this issue by giving the possibility with the new boolean argument **step** to include/exclude (if True, default being False) the events based also on the 2nd column of the events array.
Here, this will give :

```
>>> events_step = mne.pick_events(events_step, exclude = [32768], step = True)
>>> events_step[:10]
array([[222225,      0,     10],
       [222229,     10,      0],
       [223476,      0,     18],
       [223480,     18,      0],
       [224726,      0,     34],
       [224730,     34,      0],
       [237048,      0,      5],
       [237055,      5,      0],
       [238282,      0,     33],
       [238289,     33,      0]])
```

**Comment by agramfort** [comment]
I get the idea now but why calling this step? is it a name used elsewhere?

**Comment by cmoutard** [comment]
I called it step because I met this issue/need with events built with the mne.find_events' option *output = 'step'*.
If there would be other situations where the second column should be taken into account in the pick_events() function, this name would indeed not be fully suitable. But I cannot see such situations (maybe mistakenly).
We could use a more conservative name just in case, but I miss inspiration...

File: mne/event.py [hunk 1]
```diff
@@ -31,6 +32,9 @@ def pick_events(events, include=None, exclude=None):
         A event id to exclude or a list of them.
         If None no event is excluded. If include is not None
         the exclude parameter is ignored.
+    step : bool
+        If True (default is False), events have a step format.
```
Comment (agramfort) [inline mne/event.py]: add a note referring to output='step' param in find_events

File: mne/event.py [hunk 2]
```diff
@@ -31,6 +32,9 @@ def pick_events(events, include=None, exclude=None):
...
+        In this case, the two last columns are considered in inclusion/exclusion criteria.
```
Comment (agramfort) [inline mne/event.py]: make sure line is less than 80 characters

**Comment by agramfort** [comment]
also please add a test.

Also you'll need to rebase as you have a merge conflict.

thanks for clarifying the use case.

**Pull request closed without merging**